internal/cmd/perfcomp/main.go

```diff
@@ -0,0 +1,270 @@
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"go.mongodb.org/mongo-driver/v2/bson"
+	"go.mongodb.org/mongo-driver/v2/mongo"
+	"go.mongodb.org/mongo-driver/v2/mongo/options"
+	"gonum.org/v1/gonum/mat"
+)
+
+type RawData struct {
+	Info struct {
+		Project      string `bson:"project"`
+		Version      string `bson:"version"`
+		Variant      string `bson:"variant"`
+		Order        int64  `bson:"order"`
+		TaskName     string `bson:"task_name"`
+		TaskID       string `bson:"task_id"`
+		Execution    int64  `bson:"execution"`
+		Mainline     bool   `bson:"mainline"`
+		OverrideInfo struct {
+			OverrideMainline bool        `bson:"override_mainline"`
+			BaseOrder        interface{} `bson:"base_order"`
+			Reason           interface{} `bson:"reason"`
+			User             interface{} `bson:"user"`
+		}
+		TestName string                 `bson:"test_name"`
+		Args     map[string]interface{} `bson:"args"`
+	}
+	CreatedAt   interface{} `bson:"created_at"`
+	CompletedAt interface{} `bson:"completed_at"`
+	Rollups     struct {
+		Stats []struct {
+			Name     string      `bson:"name"`
+			Val      float64     `bson:"val"`
+			Metadata interface{} `bson:"metadata"`
+		}
+	}
+	FailedRollupAttempts int64 `bson:"failed_rollup_attempts"`
+}
+
+type StableRegion struct {
+	TimeSeriesInfo struct {
+		Project     string                 `bson:"project"`
+		Variant     string                 `bson:"variant"`
+		Task        string                 `bson:"task"`
+		Test        string                 `bson:"test"`
+		Measurement string                 `bson:"measurement"`
+		Args        map[string]interface{} `bson:"args"`
+	}
+	Start                  interface{}   `bson:"start"`
+	End                    interface{}   `bson:"end"`
+	Values                 []float64     `bson:"values"`
+	StartOrder             int64         `bson:"start_order"`
+	EndOrder               int64         `bson:"end_order"`
+	Mean                   float64       `bson:"mean"`
+	Std                    float64       `bson:"std"`
+	Median                 float64       `bson:"median"`
+	Max                    float64       `bson:"max"`
+	Min                    float64       `bson:"min"`
+	CoefficientOfVariation float64       `bson:"coefficient_of_variation"`
+	LastSuccessfulUpdate   interface{}   `bson:"last_successful_update"`
+	Last                   bool          `bson:"last"`
+	Contexts               []interface{} `bson:"contexts"`
+}
+
+type EnergyStats struct {
+	Benchmark    string
+	Measurement  string
+	PatchVersion string
+	StableRegion StableRegion
+	PatchValues  []float64
+	PChange      float64
+	E            float64
+	T            float64
+	H            float64
+	Z            float64
+}
+
+func main() {
+	// Connect to analytics node
+	uri := os.Getenv("perf_uri_private_endpoint")
+	if uri == "" {
+		log.Panic("perf_uri_private_endpoint env variable is not set")
+	}
+
+	version := os.Getenv("VERSION_ID")
+	if version == "" {
+		log.Panic("could not retrieve version")
```

Comment:
Add the env var that must be set to the message.


```suggestion
		log.Panic("VERSION_ID env variable is not set")
```